File: recusion/Factorial.java
public class Factorial {
    public static void main(String[] args) {
        System.out.println(recFactorial(5));
        System.out.println(itFactorial(5));
    }

    private static int recFactorial(int num) {
        if (num == 1 || num == 2) {
            return num;
        }

        return num * recFactorial(num-1);
    }

    private static int itFactorial(int num) {
        int fact = 1;
        for (int i = 2; i <= num ; i++) {
            fact = fact * i;
        }
        return fact;
    }
}

File: recusion/factorial.js
function findFactorialRecusive(num) {
    if (num === 2) {
        return num;
    }
    return num * findFactorialRecusive(num - 1);
}

function findFactorialIterative(num) {
    let fact = 1;
    for (let i = num; i > 1; i--) {
        fact = fact * i;
    }
    console.log(fact);
}

findFactorialIterative(5);

console.log(findFactorialRecusive(5));
